src/it/scala/ie/zalando/fabric/gateway/ApiValidationSpec.scala

```diff
@@ -175,19 +176,52 @@ class ApiValidationSpec
   }
 
   // Tests were timing out with the mock...
-  implicit val routeTestTimeout = RouteTestTimeout(5.seconds)
+  implicit val routeTestTimeout: RouteTestTimeout = RouteTestTimeout(5.seconds)
 
   it should "accept a stackset managed resource and create the ingress based on the response from the K8s API" in {
     synchRequest(ValidSynchRequestWithStackSetManagedServices.payload) ~> Route.seal(
       createRoutesFromDerivations(ingressDerivationLogic)) ~> check {
       val ingressii = responseAs[TestSynchResponse].ingressii
       ingressii should have length 13
       ingressii.map(_.rules.head.paths.map(_.serviceName)).foreach { backends =>
-        backends should contain allOf ("my-test-stackset-svc1", "my-test-stackset-svc2")
+        backends should contain only ("my-test-stackset-svc1", "my-test-stackset-svc2")
       }
     }
   }
 
+  it should "add version-specific hosts to ingresses for a stackset managed resource when configured" in {
+    synchRequest(ValidSynchRequestWithStackSetManagedServices.payload) ~> Route.seal(
+      createRoutesFromDerivations(new IngressDerivationChain(stackSetOperations, true))) ~> check {
+      val ingressii = responseAs[TestSynchResponse].ingressii
+
+      val mainIngressii = ingressii.filter(_.rules.map(_.host).contains("my-app.smart-product-platform-test.zalan.do"))
+      val versionedHost1 = ingressii.filter(_.rules.map(_.host).contains("my-test-stackset-svc1.smart-product-platform-test.zalan.do"))
+      val versionedHost2 = ingressii.filter(_.rules.map(_.host).contains("my-test-stackset-svc2.smart-product-platform-test.zalan.do"))
+
+      mainIngressii should have length 13
+      versionedHost1 should have length 13
+      versionedHost2 should have length 13
+
+      mainIngressii.flatMap(_.rules.map(_.paths.map(_.serviceName))).foreach { backends =>
+        backends should contain only("my-test-stackset-svc1", "my-test-stackset-svc2")
+      }
+      mainIngressii.map(_.allAnnos).foreach { annos =>
+        annos should contain ("zalando.org/backend-weights" -> Json.fromString("{\"my-test-stackset-svc1\":80,\"my-test-stackset-svc2\":20}"))
+      }
+      versionedHost1.flatMap(_.rules.map(_.paths.map(_.serviceName))).foreach { backends =>
+        backends should contain only "my-test-stackset-svc1"
+      }
+      versionedHost2.flatMap(_.rules.map(_.paths.map(_.serviceName))).foreach { backends =>
+        backends should contain only "my-test-stackset-svc2"
+      }
+      (versionedHost1++versionedHost2).map(_.allAnnos).foreach { annos =>
+        annos.keys should not contain "zalando.org/backend-weights"
+      }
+      // all ingress names should be unique
+      ingressii.map(_.name).toSet.size shouldBe ingressii.size
+    }
+  }
+
   it should "accept a stackset managed resource but create no ingress because the stackset can't be found from the K8s API" in {
```

Comment:
I'd like to see more of the Acceptance  criteria around this. For instance: 
What happens when Stacks have 0% and 100% traffic? Is there still a need for a versioned host? 
When a Stack is removed, how and when does the Versioned Host get removed?

Reply:
Discussed locally